### PR TITLE
Mixed vehicle charging demand profile

### LINKED ISSUE #152: BEV charging demand profiles created for cars only
Adapt them so that they account for all electric demand in the mobility sector.

### PR DESCRIPTION
**Pull request by SabineHaas**
Fix  #152

Changes in this PR:

- Adapted `prepare_vehicle_charging_demand()`: The pkw charging demand profiles of simBEV are mixed with a constant share in case `const_share` is not None. Formula:

`pkw_profile * (1 - const_share) + constant_profile * const_share`
        (using specified profiles, i.e. divided by their yearly sum)

- Added `pkw_share` to `demands.csv`.

todos
- [x] read const_share (or pkw_share) from file. Which file?
- [x] check whether there's a huge difference between 0.51 and 0.48 pkw_share (2050 and 2040 from BMWi) or 0.44 and 0.52 (min and max of scenarios)
- [x] check that resulting profiles sum up to 1
- [x] update file (containing const_share) on server

### CODE REVIEW

**Comment by SabineHaas** [comment]
@JL to test this PR you need to get the new `demands.csv`

File: scripts/prepare_vehicle_charging_demand.py [hunk 1]
```diff
@@ -18,14 +21,14 @@
...
 This script prepares electric vehicle charging demand profiles for the regions Berlin and
-Brandenburg. The profiles have been created before with simBEV
-(https://github.com/rl-institut/simbev). The charging strategy of simBEV data is "greedy", i.e.
-batteries are charged with maximum power until they are fully charged or removed. This script
-applies a charging strategy we refer to as "balanced" for the profiles "home" and "work" during
-specific hours (see global variables). To apply this charging strategy values between
-[`HOME_START`, `HOME_END`] and [`WORK_START`, `WORK_END`] respectively are replaced by the average
-of all these values. We assume that this is a more realistic picture of the future than a charging
-strategy "greedy".
+Brandenburg. The profiles for pkws have been created before with simBEV
```
Comment (jnnr) [inline scripts/prepare_vehicle_charging_demand.py:24]: passenger cars (German PKW)

File: scripts/prepare_vehicle_charging_demand.py [hunk 2]
```diff
@@ -35,19 +38,26 @@
...
-TS_COMMENT = "https://github.com/rl-institut/simbev"
+TS_SOURCE = "https://github.com/rl-institut/simbev"
+TS_COMMENT = (
+    "created with simBEV. contains a mix of hourly simBEV and a constant profile"
+)
 HOME_START = "15:00"  # start charging strategy "balanced" for home profile
 HOME_END = "05:00"  # end charging strategy "balanced" for home profile
 WORK_START = "06:00"  # start charging strategy "balanced" for work profile
 WORK_END = "14:00"  # end charging strategy "balanced" for work profile
 REGION_DICT = {"Berlin": "B", "Brandenburg": "BB"}
```
Comment (jnnr) [inline scripts/prepare_vehicle_charging_demand.py:49]: Could move these constants to config.settings.

File: scripts/prepare_vehicle_charging_demand.py [hunk 3]
```diff
@@ -113,6 +126,19 @@ def get_year_region_from_filename():
         # divide by total electricity demand of vehicles
         ts_total_norm = ts_total_demand / ts_total_demand.sum()
 
+        if const_share is not None:
+            # combine pkw profile and constant profile with `const_share`
+            length = len(ts_total_demand)
+            constant_ts_norm = pd.DataFrame(
+                [1 / length for i in range(length)],
```
Comment (jnnr) [inline scripts/prepare_vehicle_charging_demand.py:133]:
```suggestion
                [1/length] * length,
```

**Review by jnnr — COMMENTED** [review]
Looks good, thank you @SabineHaas! I got a few comments, but nothing big. DId not test it yet, but can do, if you want me to.

**Comment by SabineHaas** [comment]
> Looks good, thank you @SabineHaas! I got a few comments, but nothing big. DId not test it yet, but can do, if you want me to.

Thanks for your review  @jnnr . I have tested it locally - I also ran the simulations in debug mode (3 time steps). For me it's fine if you do not test it, but also nice if you find the time, of course.

File: Snakefile [hunk 4]
```diff
@@ -109,7 +109,7 @@ rule prepare_vehicle_charging_demand:
     output:
         "results/_resources/ts_load_electricity_vehicles.csv"
     shell:
```
Comment (jnnr) [inline Snakefile:111]: You could add logging. Have a look at the other rules to see how.

**Review by jnnr — APPROVED** [review]
You can merge if you want. Just for you to know: You can already set up logging, if you have another couple of minutes.

File: Snakefile (hunk 4 above)
Comment (SabineHaas) [reply Snakefile:113]: Yes, nice! Added in [5a3af6a](https://github.com/rl-institut/oemof-B3/pull/196/commits/5a3af6a958e5f916105f571c83319682306f65a9)

**Pull request merged**